doc/changelog.rst

```diff
@@ -11,6 +11,8 @@ PyMongo 4.7 brings a number of improvements including:
   :attr:`pymongo.monitoring.CommandSucceededEvent.server_connection_id`, and
   :attr:`pymongo.monitoring.CommandFailedEvent.server_connection_id` properties.
 - Fixed a bug where inflating a :class:`~bson.raw_bson.RawBSONDocument` containing a :class:`~bson.code.Code` would cause an error.
+- Replaced usage of :class:`bson.son.SON` on all internal classes and commands to dict,
```

Comment:
Let's call this out more prominently under an "Unavoidable breaking changes" section like we do here: https://github.com/mongodb/mongo-python-driver/blob/3d63eedb/doc/changelog.rst?plain=1#L291

It's also best practice to show how this breaking change could show up and what to do about it. Including example code that expects `CommandStartedEvent.command` to be SON would be good.

Reply:
https://github.com/mongodb/mongo-python-driver/pull/1498